File: Atividade_13/Mat.java
class Matematica {
    public static int fatorial(int n) {
        if (n == 0) {
            return 1;
        }
        return n * fatorial(n - 1);
    }

    public static void main(String[] args) {
        System.out.println(Matematica.fatorial(5));
    }
}

File: Atividade_13/Mat.go
package main

import "fmt"

type Matematica struct{}

func (m Matematica) Fatorial(n int) int {
	if n == 0 {
		return 1
	}
	return n * m.Fatorial(n-1)
}

func main() {
	m := Matematica{}
	fmt.Println(m.Fatorial(5))
}
